v2/sourcedb-to-spanner/src/main/java/com/google/cloud/teleport/v2/source/reader/io/jdbc/iowrapper/JdbcIoWrapper.java

```diff
@@ -277,6 +330,49 @@ private static PTransform<PBegin, PCollection<SourceRow>> getJdbcIO(
     return jdbcIO;
   }
 
+  /**
+   * Private helper to construct {@link ReadWithUniformPartitions} as per the reader configuration.
+   *
+   * @param config Configuration.
+   * @param dataSourceConfiguration dataSourceConfiguration (which is derived earlier from the
+   *     reader configuration)
+   * @param tableConfig discovered table configurations.
+   * @param sourceTableSchema schema of the source table.
+   * @return
+   */
+  private static PTransform<PBegin, PCollection<SourceRow>> getReadWithUniformPartitionIO(
+      JdbcIOWrapperConfig config,
+      DataSourceConfiguration dataSourceConfiguration,
+      SourceSchemaReference sourceSchemaReference,
+      TableConfig tableConfig,
+      SourceTableSchema sourceTableSchema) {
+
+    ReadWithUniformPartitions.Builder<SourceRow> readWithUniformPartitionsBuilder =
+        ReadWithUniformPartitions.<SourceRow>builder()
+            .setTableName(tableConfig.tableName())
+            .setPartitionColumns(tableConfig.partitionColumns())
+            .setDataSourceProviderFn(JdbcIO.PoolableDataSourceProvider.of(dataSourceConfiguration))
+            .setDbAdapter(new MysqlDialectAdapter(MySqlVersion.DEFAULT))
+            .setApproxTotalRowCount(tableConfig.approxRowCount())
+            .setRowMapper(
+                new JdbcSourceRowMapper(
+                    config.valueMappingsProvider(),
+                    sourceSchemaReference,
+                    sourceTableSchema,
+                    config.shardID()))
+            .setWaitOnSignals(config.waitOnSignals())
+            // TODO(vardhanvthigle): if index is not of the type of a single auto incrementing key,
+            // don't set this.
+            .setSplitStageCountHint(4L)
```

Comment:
what is this hint ?

Reply:
This is the number of stages we provision to split the ranges.

We have java doc [at the source](https://github.com/GoogleCloudPlatform/DataflowTemplates/pull/1657/files/2f1562c543def759334e7abc570222937c0b67fe#diff-63967ef4a0e72c5ff642b8a7351e3eca71bf77e9d9929eabd70ec91b4d34aea2R125)

This is not a compulsory parameter, it gets auto inferred, but given that we are dealing with only auto incrementing keys till we get in strings, we don't need a large split depth even for tall tables.

I will add more details in the comment in JDBCIOWrapper too.